src/coreclr/unitygc/unitygc.cpp

```diff
@@ -0,0 +1,901 @@
+#include "common.h"
+#if defined(TARGET_WINDOWS)
+#include "windows.h"
+#endif
+#include "gcenv.h"
+
+#define UGC_ALIGN(size, align) \
+    (((size)+((align)-1)) & ~((align)-1))
+
+static const char* sUnityGC = "UnityGC";
+
+#ifdef unitygc_EXPORTS
+#define UNITYGC_EXPORT __declspec(dllexport)
+#else
+#endif
+
+class GCHeap : public IGCHeap
+{
+public:
+    /*
+    ===========================================================================
+    Hosting APIs. These are used by GC hosting. The code that
+    calls these methods may possibly be moved behind the interface -
+    today, the VM handles the setting of segment size and max gen 0 size.
+    (See src/vm/corehost.cpp)
+    ===========================================================================
+    */
+
+    // Returns whether or not the given size is a valid segment size.
+    virtual bool IsValidSegmentSize(size_t size)
+    {
+        assert(0);
+        return true;
+    }
+
+    // Returns whether or not the given size is a valid gen 0 max size.
+    virtual bool IsValidGen0MaxSize(size_t size)
+    {
+        assert(0);
+        return true;
+    }
+
+    // Gets a valid segment size.
+    virtual size_t GetValidSegmentSize(bool large_seg = false)
+    {
+        assert(0);
+        return 0;
+    }
+
+    // Sets the limit for reserved virtual memory.
+    virtual void SetReservedVMLimit(size_t vmlimit)
+    {
+        assert(0);
```

Comment:
Just want to clarify we want to assert if these functions are called? The comments seem disconnected from the behavior.

Reply:
Yes, I am asserting in all the unimplemented methods and handling as we hit them. The comments are from the header which defines the interface we implement, stating what each method should do.

Reply:
Okay cool makes sense then.